File: 881-boats-to-save-people/881-boats-to-save-people.cpp
class Solution {
public:
    
    int numRescueBoats(vector<int>& people, int limit) {    
        //Sort all of people 
        sort(people.begin(),people.end());
        
        int i = 0;
        int j = people.size() -1;
        
        //Initially count value is 0
        int count = 0;
        
        while(i <= j)
        {
            //This will represent that add light weight people with heavy weight people
            if(people[i] + people[j] <= limit)
                
                i++;
                j--;
                count++;
        }
        
        return count;
        
    }
};

File: 881-boats-to-save-people/881-boats-to-save-people.java
class Solution {
    
    public int numRescueBoats(int[] people, int limit) {    
        Arrays.sort(people);
        
        int i=0;
        int j=people.length - 1;
        
        int answer = 0;
        
        while(i <= j)
        {
            if(people[i] + people[j] <= limit)
            i++;
            j--;
            answer++;
        }
        return answer;
    }
}